src/server/components/lists/searches/translators-interpreters.ts

```diff
@@ -0,0 +1,133 @@
+import { Request, Response } from "express";
+import { ROWS_PER_PAGE, getPaginationValues } from "server/models/listItem/pagination";
+import { DEFAULT_VIEW_PROPS } from "../constants";
+import {
+  getServiceLabel,
+  getAllRequestParams,
+  removeQueryParameter,
+  getParameterValue,
+  queryStringFromParams, parseListValues
+} from "../helpers";
+import { QuestionName } from "../types";
+import { getCSRFToken } from "server/components/cookies/helpers";
+import { TranslatorInterpreterListItem } from "server/models/listItem/providers";
+import { languages, translationInterpretationServices } from "server/services/metadata";
+import { TranslatorInterpreterListItemGetObject } from "server/models/types";
+import { cleanLanguagesProvided } from "server/models/listItem/providers/helpers";
+import { camelCase } from "lodash";
+
+export const translatorsInterpretersQuestionsSequence = [
+  QuestionName.readNotice,
+  QuestionName.country,
+  QuestionName.region,
+  QuestionName.servicesProvided,
+  QuestionName.languagesProvided,
+  QuestionName.languagesSummary,
+  QuestionName.translationSpecialties,
+  QuestionName.interpreterServices,
+  QuestionName.interpreterTranslationServices,
+  QuestionName.readDisclaimer,
+];
+
+export async function searchTranslatorsInterpreters(
+  req: Request,
+  res: Response
+): Promise<void> {
+  const params = getAllRequestParams(req);
+  const { serviceType, country, region, print = "no", languagesProvided } = params;
+  let languageNamesProvided, serviceNamesProvided;
+  let servicesProvided = parseListValues("servicesProvided", params);
+  if (servicesProvided != null) {
+    servicesProvided = servicesProvided.map((service) => service.toLowerCase());
+  }
+  let translationSpecialties = parseListValues("translationSpecialties", params);
+  if (translationSpecialties != null) {
+    translationSpecialties = translationSpecialties.map((service) => service.toLowerCase());
+  }
+  let interpreterServices = parseListValues("interpreterServices", params);
+  if (interpreterServices != null) {
+    interpreterServices = interpreterServices.map((service) => service.toLowerCase());
+  }
+
+  let { page = "1" } = params;
+  page = page !== "" ? page : "1";
```

Comment:
yep, that's correct.  It's a shorthand way of defaulting a value when deconstructing.

Reply:
@RichardBray I would recommend not using `Boolean` whenever a number is involved. `Boolean(thing)` is used in javascript to detect whether or not a thing exists as a shortcut. Same with `!!thing`. The intent won't  be clear in a few months time.. you could be checking for an empty string (falsey), or a number (0, 1), undefined, null..  Only use these when you're dealing with strings or objects

Reply:
hm... perhaps I would actually just use `||` in this case? 

fwiw - destructuring default values only deals with undefined values. if it is defined as an empty string, it won't default. 

```const page = page.params || "1"```

Reply:
That's a good point @jenbutongit I guess the number could be `0` which would be false. I guess I assumed `params.page` would be a string but it could be a number 🤷